README.md

```diff
@@ -70,15 +119,279 @@ solidity: |-
+
+e.g. making it trivial to submit transactions. No coding required, just open up Postman and send in a trivial piece of YAML copied from a README like this one.
+
+In an Enterprise context, the availability of simple standardized interfaces like HTTP and Kafka means applications and integration tools (a common ESB or other EAI tooling) can be hooked up trivially. No need to insert complex code libraries into applications,
+many of which are LGPL licensed.
+
+## Why Messaging?
+
+So you ask, if the goal is simplicity, why not just put the simple HTTP API in front of JSON/RPC and be done with it?
+
+There are some challenges in Enterprise grade Blockchain solutions (particularly in high throughput permissioned/private chains) that cannot be solved by a stateless HTTP bridging layer alone.
+
+So for Kaleido, we started with a robust Messaging tier and layered the HTTP interface on top.
+
+## The asynchronous nature of Ethereum transactions
+
+Ethereum transactions can take many seconds or minutes (depending on the backlog and block period) from submission until they make it into a block, and a receipt is available. Each individual node in Ethereum provides a built-in simple pool where transactions can be pooled while waiting to enter a block.
+
+Connecting a synchronous blocking HTTP interface directly to an inherently asynchronous system like this can cause problems. The HTTP requester times out waiting for a response, but has no way to know if it should retry or if the transaction will eventually succeed. It cannot cancel the request.
+
+Providing a Messaging layer with at-least-once delivery and message ordering, allows the asynchronous nature of Ethereum to be reflected back to the remote application.
+
+Applications that have their own state stores are able to communicate over Messaging / Kafka with trivially simple JSON payloads to stream transactions into a scalable set of Ethereum nodes, and process the replies as they occur. The application can scale horizontally. Applications can also be decoupled from the Ethereum network with an integration technology like an Enterprise Service Bus (ESB).
+
+When spikes in workload occur that create a large queue of transactions that need to be fed into the Ethereum network at a lower rate, the kaleido-io/ethconnect bridge feeds them in at an optimal rate.
+
+### Ethereum Webhooks and the REST Receipt Store (MongoDB)
+
+Another key goal of having a robust Messaging layer under the covers is that the most trivial application can send messages into Ethereum reliably.
+
+- `POST` a [trivial YAML/JSON payload](#yaml-to-submit-a-transaction)
+  - to `/hook` to complete once the message is confirmed by Kafka (0.5s - tunable)
+  - to `/fasthook` to complete immediately when the message is sent to Kafka
+- Get an _immediate_ receipt
+   - No need to wait many seconds for a block to be cut
```

Comment:
As someone who has worked a teeny bit with ethereum the idea of an _immediate_ receipt is a little scary to me. This sounds like Kafka just says OK I got your stuff, which is very similar to sending a transaction into a node. The long, short of it is I don't _believe_ in this immediate receipt.

I imagine what this means is that Kafka confirms _it_ has received my transaction, but I will receive a webhook at a later date. Which is exciting! No retry polling (which you talk about later). It might be good to hit that point here instead of leaving it for later.